### PR TITLE
Tasks for 02.07

### PR DESCRIPTION
**Pull request by pavlentiysv**
solving 6 tasks for 02-04.07

### CODE REVIEW

File: js/02.07/task5.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+function debounce(fn, timeOut) {
+  if (timer) {
+    clearTimeout(timer);
+  }
+  timer = setTimeout(fn, timeOut);
+}
+
+let timer;
```
Comment (dlos-hiqo) [inline js/02.07/task5.js]: Please rewrite using closure

File: js/02.07/task4.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+function curry(fn) {
```
Comment (dlos-hiqo) [inline js/02.07/task4.js]: Please rewrite using .bind

File: js/02.07/task6.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+function memoize(fn) {
```
Comment (dlos-hiqo) [inline js/02.07/task6.js]: Please rewrite using closure

File: js/02.07/task6.js [hunk 4]
```diff
@@ -1,12 +1,16 @@
 function memoize(fn) {
-  fn.memoCache = {};
+  const memoCache = {};
   return (...args) => {
```
Comment (Pozharitskiy) [inline js/02.07/task6.js]: lol I have the same solution!
Rewrite it please

Comment (pavlentiysv) [reply js/02.07/task6.js]: oh sh*t i'm sorry

**Pull request merged**